CMakeLists.txt

```diff
@@ -109,9 +109,15 @@ elseif("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m32")
 endif()
 
-find_package(hsa-runtime64 REQUIRED
-   PATHS  /opt/rocm
+if (DEFINED ROCM_HOME)
```

Comment:
>> No need to add extra variables
What is the optimization of you get out of "No need to add extra variables" or what is the downside? The CMAKE_PREFIX_PATH and other cmake defined paths have side-effects and using those variables to get what user wants cmake to do is incorrect application of those variables, which leads to more headaches.
Read https://cmake.org/cmake/help/v3.0/variable/CMAKE_INSTALL_PREFIX.html
CMAKE_PREFIX_PATH is to specify destination dir. That is not the purpose of the change here.

Reply:
What you are referring here is CMAKE_INSTALL_PREFIX.

CMAKE_PREFIX_PATH gives a different purpose.

https://cmake.org/cmake/help/v3.0/variable/CMAKE_PREFIX_PATH.html

Reply:
Ok, my mistake. But please see my comments on line 170+ as to avoid "deduction logic" and providing user-defined parameter to use with runpath, etc.
My changes allow DevOps and other cmake experts to continue to use CMAKE_PREFIX_PATH to do a find as well as simple users who want to specify the lcoation to use, and use that in runpath.